redis/tests/test_sentinel.rs

```diff
@@ -162,32 +147,44 @@ fn test_sentinel_client() {
 
     let mut master_con = master_client.get_connection().unwrap();
 
-    // Write commands would go to the primary nodes
-    redis::cmd("SET")
-        .arg("{x}key1")
-        .arg(b"foo")
-        .execute(&mut master_con);
-    redis::cmd("SET")
-        .arg(&["{x}key2", "bar"])
-        .execute(&mut master_con);
+    check_master_role(&mut master_con);
+
+    let sentinel = context.sentinel_mut();
+    let master_client = sentinel.master_for("master1").unwrap();
 
     for _ in 0..20 {
         let mut replica_con = replica_client.get_connection().unwrap();
 
-        assert_eq!(
-            redis::cmd("MGET")
-                .arg(&["{x}key1", "{x}key2"])
-                .query(&mut replica_con),
-            Ok(("foo".to_string(), b"bar".to_vec()))
-        );
+        check_replica_role_and_master(&mut replica_con, &master_client);
     }
 }
 
 #[cfg(feature = "aio")]
 pub mod async_tests {
-    use redis::{sentinel::SentinelClient, RedisError, RedisResult};
+    use redis::{aio::Connection, sentinel::SentinelClient, Client, RedisError};
 
-    use crate::support::*;
+    use crate::{check_replica_info_result, parse_replication_info, support::*};
+
+    async fn async_check_master_role(conn: &mut Connection) {
+        let info: String = redis::cmd("INFO")
+            .arg("REPLICATION")
+            .query_async(conn)
+            .await
+            .unwrap();
+
+        let info_map = parse_replication_info(&info);
```

Comment:
why not use `check_master_role`?

Reply:
I hadn't extracted this part as a separate function yet, the `check_master_role` didn't use async. But while renaming with your suggestions, I've also extracted this part to a separate function.